src/custom_types.rs

```diff
@@ -0,0 +1,77 @@
+use lsp_data::Range;
```

Comment:
I'm not clear what the difference between these types and the rls-data types is? Ideally we'd just use one set of types. If that is not possible, then this setup seems fine.